pkg/cfn/manager/create_tasks.go

```diff
@@ -32,7 +42,7 @@ func (c *StackCollection) NewTasksToCreateClusterWithNodeGroups(nodeGroups []*ap
 
 	if nodeGroupTasks.Len() > 0 {
 		nodeGroupTasks.IsSubTask = true
-		tasks.Append(nodeGroupTasks)
+		postClusterTasks.Append(nodeGroupTasks)
```

Comment:
Don't these need to be appended to `tasks` instead of `postClusterTasks` so they don't get lost given that they get added to `tasks` on line 31?

Reply:
i think it's ok since `postClusterTasks` is a pointer to my `Tree`, I can `Append` to it even after I've `Append`ed it to `tasks` on line 31. and in the other branch  line 32-34, `postClusterTasks == tasks`.
but it's actually hard to read as it is, I'll improve it

Reply:
hopefully it's a bit easier to see what's going on now

Reply:
Right. For some reason I assumed that the tasks.Append() would append the tasks from one tree into the other but it appends the tree directly obviously. Thanks.

Reply:
Both ways work for me, leave it as you prefer :+1: